crates/host/src/oci.rs

```diff
@@ -260,4 +279,10 @@ impl Fetcher {
             .await
             .with_context(|| format!("failed to read `{}`", path.display()))
     }
+
+    /// Used to set additional CA paths that will be used as part of fetching components and providers
+    pub fn with_additional_ca_paths(mut self, additional_ca_paths: &Vec<PathBuf>) -> Self {
```

Comment:
```suggestion
    pub fn with_additional_ca_paths(mut self, additional_ca_paths: &[impl AsRef<Path>]) -> Self {
```

NIT: Requires some more changes but, might be nice to have the flexibility

Reply:
I'm going to leave this one for someone else who has the patience with the borrow checker, I can't figure out how to make this work nicely. I'm happy to open up a follow-up issue for tracking it if this needs to be done.

Reply:
All good thank you for trying -- I'll do this!